stdlib/REPL/src/LineEdit.jl

```diff
@@ -1272,8 +1272,30 @@ function write_prompt(terminal, p::Prompt)
     return width
 end
 
+# On Windows, when launching external processes, we cannot control what assumption they make on the
+# console mode. We thus forcibly reset the console mode at the start of the prompt to ensure they do
+# not leave the console mode in a corrupt state.
+if Sys.iswindows()
+function _console_mode()
+    hOutput = ccall(:GetStdHandle, stdcall, Int32, (Int32,), -11) # STD_OUTPUT_HANDLE
+    dwMode = Ref{UInt32}(0)
+    ccall(:GetConsoleMode, stdcall, Int32, (Int32, Ptr{Nothing}), hOutput, dwMode)
+    dwMode[]
+end
+const default_console_mode = _console_mode()
```

Comment:
should not be a build-time constant

Reply:
Good point! I changed to be a reference that get's set  the first time the REPL is printed at runtime and then is cached perpetually.